src/mlpack/methods/ann/layer/lstm_impl.hpp

```diff
@@ -372,7 +370,7 @@ void LSTMType<MatType>::Gradient(
   offset += cell2GateOutputWeight.n_elem;
 
   // cell2GateForgetWeight and cell2GateInputWeight gradients.
-  if (this->HasPreviousStep()) // TODO: fix convention here
+  if (this->HasPreviousStep())
   {
     gradient.submat(offset, 0, offset + cell2GateForgetWeight.n_elem - 1, 0) =
```

Comment:
@rcurtin Instead of `gradient.submat(offset, 0, offset + cell2GateForgetWeight.n_elem - 1, 0)`, maybe it would be more readable (and hence more maintainable) to use `gradient(offset, 0, arma::size(cell2GateForgetWeight.n_elem,1))`.

Armadillo submatrices can take `size(n_rows, n_cols)` arguments: http://arma.sourceforge.net/docs.html#submat